docs/src/content/docs/get-started/url-shortner.mdx

```diff
@@ -0,0 +1,306 @@
+
+    ```go
+    package services
+
+    import (
+        "fmt"
+        "url-shortner-dicedb/models"
+        "url-shortner-dicedb/repository"
+
+        "github.com/google/uuid"
+    )
+
+    func CreateShortURLService(longURL string) (string, error) {
+        id := uuid.New().String()[:8] // Shorten ID for URL
+
+        url := models.URL{
+            ID:       id,
+            LongURL:  longURL,
+            ShortURL: "http://localhost:8080/" + id,
+        }
+
+        if err := repository.SaveURL(url); err != nil {
+            fmt.Println(err)
+            return "", err
+        }
+
+        return url.ShortURL, nil
+    }
+
+    func GetOriginalURLService(id string) (string, error) {
+        url, err := repository.FindURLByID(id)
+        if err != nil {
+            return "", err
+        }
+
+        return url.LongURL, nil
+    }
+
+    ```
+
+    - The `CreateShortURLService` function generates a unique ID for each URL and stores it using `SaveURL`.
+
+4. `handlers/url_handler.go`: Defines HTTP endpoints for creating and redirecting URLs.
+
+    ```go
+    package handlers
+
+    import (
+        "net/http"
+        "url-shortner-dicedb/models"
+        "url-shortner-dicedb/services"
+        "url-shortner-dicedb/utils"
+
+        "github.com/gin-gonic/gin"
+    )
+
+    func RegisterRoutes(router *gin.Engine) {
+        router.POST("/shorten", CreateShortURL)
+        router.GET("/:id", RedirectURL)
+    }
+
+    func CreateShortURL(c *gin.Context) {
+        var requestBody models.URL
+        if err := c.ShouldBindJSON(&requestBody); err != nil {
+            utils.RespondWithError(c, http.StatusBadRequest, "Invalid request")
+            return
+        }
+
+        shortURL, err := services.CreateShortURLService(requestBody.LongURL)
+        if err != nil {
+            utils.RespondWithError(c, http.StatusInternalServerError, "Error creating short URL")
+            return
+        }
+
+        utils.RespondWithJSON(c, http.StatusCreated, gin.H{"short_url": shortURL})
+    }
+
+    func RedirectURL(c *gin.Context) {
+        id := c.Param("id")
+        longURL, err := services.GetOriginalURLService(id)
+        if err != nil {
+            utils.RespondWithError(c, http.StatusNotFound, "URL not found")
+            return
+        }
+
+        c.Redirect(http.StatusFound, longURL)
+    }
+    ```
+
+    - `CreateShortURL` handles POST requests to generate a short URL.
+    - `RedirectURL` redirects to the original URL based on the short ID.
+
+5. `utils/response.go`: Provides helper functions for consistent JSON responses.
+
+    ```go
+    package utils
+
+    import (
+        "github.com/gin-gonic/gin"
+    )
+
+    func RespondWithError(c *gin.Context, code int, message string) {
+        c.JSON(code, gin.H{"error": message})
+    }
+
+    func RespondWithJSON(c *gin.Context, code int, payload interface{}) {
+        c.JSON(code, payload)
+    }
+    ```
```

Comment:
no need to have this in a separate package/file. the actual code these functions encapsulate is simpler to understand at first glance as compared to hiding them behind an abstraction